packages/analytics-browser/src/session-manager.ts

```diff
@@ -10,23 +10,39 @@ export class SessionManager implements ISessionManager {
   storageKey: string;
   sessionTimeout: number;
   cache: UserSession;
+  // Used for testing
+  ready: Promise<void>;
 
   constructor(private storage: Storage<UserSession>, options: SessionManagerOptions) {
     this.storageKey = getStorageKey(options.apiKey);
     this.sessionTimeout = options.sessionTimeout;
-    this.cache = this.storage.get(this.storageKey) ?? {
-      optOut: false,
-    };
+    this.cache = {} as UserSession;
+    // Load the cache asynchronously. Values set before the cache loads are
+    // preferred over storage.
+    this.ready = this.storage.get(this.storageKey).then((userSession) => {
+      // Strip existing but undefined fields from cache, otherwise spreading
+      // will overwrite a value with undefined.
+      Object.keys(this.cache).forEach((key) => {
+        if (typeof this.cache[key as keyof UserSession] === 'undefined') {
+          delete this.cache[key as keyof UserSession];
+        }
+      });
+      this.cache = { ...userSession, ...this.cache } ?? { optOut: false };
+    });
```

Comment:
This was a little weird, tests were failing due to the object spread keeping values explicitly set to `undefined` over values that were set with real values.

Happy to make changes here, not sure what the best course of action is.